### ISSUE TITLE
Failed Semver parsing in 

### ISSUE THREAD

**Issue by pnegahdar** [romejs/rome#295]
Description:

    Cannot parse version ` "^4.0.0 || ^4.0.0-beta.* || ^3.8.1 || ^3.0.0 || ^3.0.0-rc.0 || ^2.1.0 || ^1.13.1"` in package.json. 

## Steps To Reproduce
```
 node_modules/hard-source-webpack-plugin/package.json:82:38 parse/semver 

  ✖ Invalid version qualifier part

    80 │     "standard-version": "^4.4.0",
    81 │     "style-loader": "^0.14.0",
  > 82 │     "webpack": "^4.0.0 || ^4.0.0-beta.* || ^3.8.1 || ^3.0.0 || ^3.0.0-rc.0 || ^2.1.0 || ^1.13.1",
       │                                       ^
    83 │     "webpack-cli": "^3.0.0",
    84 │     "webpack-isomorphic-tools": "^3.0.0",

```

It fails to parse this part: `^4.0.0-beta.*` 

It isn't clear to me if this is this is valid or invalid.

**Comment by sebmck** [comment]
I don't think that is technically valid but if npm/yarn handles it silently then we should too. We have a "loose" mode that we enable for `package.json` files found in `node_modules` that has some fallbacks for these cases. For regular `package.json` files in your project then we validate them more strictly.

**Comment by sebmck** [comment]
Fixed in #705, thank for reporting @pnegahdar!
